web/components/contract/user-positions-table.tsx

```diff
@@ -79,28 +76,24 @@ export const BinaryUserPositionsTable = memo(
       return [positiveProfitPositions, negativeProfitPositions.reverse()]
     }, [contractMetricsByProfit])
 
-    const [livePositionsLimit, setLivePositionsLimit] = useState(100)
     const positions =
-      useContractMetrics(contractId, livePositionsLimit, outcomes) ??
-      props.positions
+      useRealtimeContractMetrics(contractId, outcomes) ?? props.positions
 
     const yesPositionsSorted =
       sortBy === 'shares' ? positions.YES ?? [] : positiveProfitPositions
     const noPositionsSorted =
       sortBy === 'shares' ? positions.NO ?? [] : negativeProfitPositions
     useEffect(() => {
-      // Let's use firebase here as supabase can be slightly out of date, leading to incorrect counts
-      getTotalContractMetricsCount(contractId).then(setTotalPositions)
-
       Promise.all([
-        getContractMetricsYesCount(contractId),
-        getContractMetricsNoCount(contractId),
-      ]).then(([yesCount, noCount]) =>
+        getContractMetricsOutcomeCount(contractId, 'yes', db),
+        getContractMetricsOutcomeCount(contractId, 'no', db),
+      ]).then(([yesCount, noCount]) => {
         setShareholderStats({
           yesShareholders: yesCount,
           noShareholders: noCount,
         })
-      )
+        setTotalPositions?.(yesCount + noCount)
+      })
```

Comment:
If the `useRealtimeContractMetrics` actually includes all user positions, then it might include these counts, so we could delete this.